drivers/adxl345/adxl345.c

```diff
@@ -43,6 +44,12 @@ int adxl345_init(adxl345_t *dev, const adxl345_params_t* params)
     /* get device descriptor */
     dev->params= *params;
```

Comment:
noo, actually what I meant is quite simple: instead of having the device descriptor look like
```c
typedef struct {
    adx1345_params_t params;
    ...
} adx1345_t;
```
you could do something like
```c
typedef struct {
    i2c_t bus,
    uint8_t addr,
    adx1345_params_t *p,
    ....
} adx1345_t;
```
In this case, you would only copy the 'often/periodically' used values into RAM for faster access, while you access the rest of the config params indirectly from ROM (we assume the configuration to be placed in ROM...).